comit/src/halight.rs

```diff
@@ -148,3 +148,11 @@ pub struct Settled {
 
 #[derive(Debug, Clone, Copy, PartialEq)]
 pub struct Cancelled;
+
+/// Which layer 1 network we are running on (implicitly refers to the Bitcoin)
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub enum Network {
+    Mainnet,
+    Testnet,
+    Regtest,
+}
```

Comment:
I would prefer to keep it modeled as `ledger::Lightning` but if you don't want to have the enum, you can have a field with a Bitcoin Network maybe?

Reply:
This change of model is the main concept of this draft PR. The reason I'm tackling this is explained in the PR message, can you explain _why_ you want to keep it as is and not unify all our models as `protocol::Type` instead of having half like that and half like this one `type::Protocol`?